lib/dependabot/update_checkers/java_script/npm_and_yarn/version_resolver.rb

```diff
@@ -318,13 +318,19 @@ def newly_broken_peer_reqs_from_dep
               select { |dep| dep[:requiring_dep_name] == dependency.name }
           end
 
+          def lockfiles_for_path(lockfiles:, path:)
+            lockfiles.select do |lockfile|
+              File.dirname(lockfile.name) == File.dirname(path)
```

Comment:
Make this a one-liner?

Reply:
Also, don't see why it should take a `lockfiles` argument - could get that from state.